src/Features/Core/Portable/DocumentHighlighting/AbstractDocumentHighlightsService.cs

```diff
@@ -14,13 +14,48 @@
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Editor.Implementation.ReferenceHighlighting
+namespace Microsoft.CodeAnalysis.DocumentHighlighting
 {
-    internal abstract class AbstractDocumentHighlightsService : IDocumentHighlightsService
+    internal abstract partial class AbstractDocumentHighlightsService : IDocumentHighlightsService
     {
         public async Task<ImmutableArray<DocumentHighlights>> GetDocumentHighlightsAsync(
             Document document, int position, IImmutableSet<Document> documentsToSearch, CancellationToken cancellationToken)
         {
+            var (succeeded, highlights) = await GetDocumentHighlightsInRemoteProcessAsync(
```

Comment:
❔ How long does failure take if someone e.g. kills the remote process? Is failure a scenario we want to latch on or retry?

💭 In OOP scenarios, this method appears to be called by both the in-process component and the out-of-process component. It would be good to verify the out-of-process component takes a fast path through the "failure" scenario since it will only ever need to call `GetDocumentHighlightsInCurrentProcessAsync`.